### PR TITLE
Update to go 1.20 min

### PR DESCRIPTION
**Pull request by smallhive**

### CODE REVIEW

**Review by roman-khimov — COMMENTED** [review]
`internal/neofs/tree.go` can be simplified in `strings.Trim*` functions, although `CutPrefix` is not exactly perfect here, it's mostly that some checks are not needed.

`api/layer/encryption/encryption.go` can be improved to use `bytes.Clone` instead of open-coded copying.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -18,10 +18,11 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Set up Go
-        uses: actions/setup-go@v4
+        uses: actions/setup-go@v5
         with:
           cache: true
           go-version: '1.22'
```
[the remark this answers is not in the document]
Comment (smallhive) [reply .github/workflows/tests.yml]: Fixed

**Comment by roman-khimov** [comment]
https://github.com/nspcc-dev/neofs-s3-gw/pull/933#pullrequestreview-1906053457

**Comment by smallhive** [comment]
`encryption.go` updated. `CutPrefix` adds the second returning parameter, it is mostly unconvinient

File: api/layer/encryption/encryption.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func NewParams(key []byte) (*Params, error) {
 	}
 	var p Params
 	p.customerKey = make([]byte, aes256KeySize)
```
Comment (roman-khimov) [inline api/layer/encryption/encryption.go:73]: So why do we need this then?

Comment (smallhive) [reply api/layer/encryption/encryption.go]: Obviously not required, updated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**